datatypes/src/collections/data_collection.rs

```diff
@@ -16,6 +16,7 @@ impl<'i> IntoGeometryOptionsIterator<'i> for DataCollection {
     }
 }
 
+#[allow(clippy::into_iter_without_iter)] // we provide `.geometries()` instead
```

Comment:
why can we not use Iter there?

Reply:
There are multiple ways to iterate over a collection. This, `.geometries()`, means iterating over geometries only. But there are ways to iterate over other columns with `.data()`, for instance, or to get time instances. Thus, it does not make sense to have `.iter()` point to exactly this iterator.

Reply:
can we make `.iter()` generic over the return type?

Reply:
I have two thoughts on this:

1. I don't think so, at least not so easily. We would need to put the iter-Impl into a trait that each of the return types must implement. So the type would know how the FeatureCollection would iterate over itself. Should be possible, but It could get blind doing this.
2. This would not be ergonomic at all. The caller would need to very specific typing instead of just calling .iter.
3. I doubt that it would make the linter happy. Can it recognize, that there is somewhat an `.iter()` method for this? I guess `.geometries()` is better suited for this, as it is now.

We could remove IntoIter and just provide `.geoemetries`, but I don't see the huge advantage.